### PR TITLE
Make sure that field collapsing supports field aliases.

### PR DESCRIPTION
**Pull request by jtibshirani**
This component of the search request wasn't covered by the original field alias PR.

Addresses #32623.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/collapse/FieldCollapseIT.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.elasticsearch.index.query.InnerHitBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import static org.elasticsearch.index.query.QueryBuilders.matchAllQuery;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertNoFailures;
+
+public class FieldCollapseIT extends ESIntegTestCase {
```
Comment (jtibshirani) [inline server/src/test/java/org/elasticsearch/search/collapse/FieldCollapseIT.java]: I looked into creating a unit test, but for just adding this single case, it was a lot cleaner to write an integration test.

**Review by jimczi — COMMENTED** [review]
I left one comment, thanks @jtibshirani

File: server/src/test/java/org/elasticsearch/search/collapse/FieldCollapseIT.java (hunk 1 above)
Comment (jtibshirani) [reply server/src/test/java/org/elasticsearch/search/collapse/FieldCollapseIT.java]: I find REST tests more difficult to work with in terms of debugging, so I've been preferring Java integration tests unless I'm actually testing the REST spec. I'm happy to switch this over though, if writing a REST test is more standard.

**Comment by jtibshirani** [comment]
Thanks @jimczi, I think it's ready for another look.

**Review by jimczi — APPROVED** [review]

**Comment by jtibshirani** [comment]
@elasticmachine run packaging tests

**Pull request merged**